legacyPost/src/main/java/com/hamdan/forzenbook/legacy/post/LegacyPostActivity.kt

```diff
@@ -1,33 +1,196 @@
+                postLoadIndicator.isVisible = false
+                if (showBottomButton) {
+                    postChooseImage.isVisible = true
+                    scrollPost.isVisible = false
+                    postChosenImage.isVisible = false
+                    postChangeImage.isVisible = false
+                } else {
+                    postChooseImage.isVisible = false
+                    scrollPost.isVisible = false
+                    postChosenImage.isVisible = true
+                    postChangeImage.isVisible = true
+                }
+            }
+            val showTextContent = {
+                postLoadIndicator.isVisible = false
+                postChooseImage.isVisible = false
+                scrollPost.isVisible = true
+                postChosenImage.isVisible = false
+                postChangeImage.isVisible = false
+            }
+            val loadingContent = {
+                postLoadIndicator.isVisible = true
+                postChooseImage.isVisible = false
+                scrollPost.isVisible = false
+                postChosenImage.isVisible = false
+                postChangeImage.isVisible = false
+            }
+            val errorContent = {
+                postLoadIndicator.isVisible = false
+                postChooseImage.isVisible = false
+                scrollPost.isVisible = false
+                postChosenImage.isVisible = false
+                postChangeImage.isVisible = false
+            }
+            postLayoutToolBar.inflateMenu(R.menu.post_menu)
+            postLayoutToolBar.setOnMenuItemClickListener { item ->
+                when (item.itemId) {
+                    R.id.postSendIcon -> {
+                        postViewModel.sendPostClicked(this@LegacyPostActivity)
+                        true
+                    }
+
+                    else -> {
+                        false
+                    }
+                }
+            }
+            pillLeftFrame.isEnabled = true
+            textToggle.isChecked = true
+            pillRightFrame.isEnabled = false
+            imageToggle.isChecked = false
+            postPill.setOnClickListener {
+                setPill(pillToggled)
+                pillToggled = !pillToggled
+                val currentFocus = this@LegacyPostActivity.currentFocus
+                if (currentFocus != null) {
+                    val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+                    imm.hideSoftInputFromWindow(currentFocus.windowToken, 0)
+                }
+            }
+
+            postChangeImage.setOnClickListener {
+                launchGalleryImageGetter(launcher)
+            }
+            postChooseImage.setOnClickListener {
+                launchGalleryImageGetter(launcher)
+            }
+
+            postViewModel.viewModelScope.launch(Dispatchers.IO) {
+                postViewModel.state.collect { state ->
```

Comment:
Other than setting up listeners this method should be the only method of view state updating

Reply:
You are definitely right, I got lost in the sauce of trying to make it easier to change what the view looked like by using the lambdas.